code/Modules/Items/Carriables/ItemTrashBag.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Sandbox;
+using ssl.Modules.Items.Data;
+using ssl.Player;
+
+namespace ssl.Modules.Items.Carriables
+{
+    public partial class ItemTrashBag : Item
+    {
+        private const int InventorySize = 6;
+        
+        public ItemTrashBag()
+        {
+        }
+
+        public ItemTrashBag(ItemData data) : base(data)
+        {
+            Content = new Inventory(InventorySize);
+        }
+        
+        public Inventory Content { get; private set; }
+
+        /// <summary>
+        /// Using the trash bag on an item on ground will add it to the trash bag
+        /// </summary>
+        public override void UseOn(MainPlayer player)
```

Comment:
I like the idea of let the bag do stuff and not the opposite. Maybe with trhe mop i have code the wrong way with mop and not items have logic.

Maybe the item UseOn(ISelectable s) and UsedBy(MainPlayer p)

What do you think ?

Reply:
Yeah UsedBy is more logical I think, we could only notice this reying to implement an item lol
Also, we should evolve this function to allow the check of different input and not just only the "use" button

Reply:
done in other mr